packages/browser-wallet/src/popup/shared/Form/Slider/Slider.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from 'react';
+import RcSlider from 'rc-slider';
+import clsx from 'clsx';
+import { noOp, toFixed, valueNoOp } from 'wallet-common-helpers';
+import { CommonFieldProps } from '../common/types';
+import { InlineInput } from '../InlineInput';
+
+interface Props extends CommonFieldProps {
+    min: number;
+    max: number;
+    step: number;
+    unit?: string;
+    value: number | undefined;
+    onChange?(value: number | number[] | undefined): void;
+    onBlur?(): void;
+    className: string;
+    name: string;
+}
+
+export default function Slider({
+    min,
+    max,
+    step,
+    label,
+    unit = '',
+    onChange = noOp,
+    onBlur = noOp,
+    value,
+    className,
+    name,
+}: Props) {
+    const [innerValue, setInnerValue] = useState<number | undefined>(value);
+    const isInvalid = innerValue !== undefined && (Number.isNaN(innerValue) || innerValue > max || innerValue < min);
+
+    const allowFractions = step < 1 && step > 0;
+    const ensureDigits = allowFractions ? step.toString().split('.')[1].length : undefined;
+    const formatNumber = ensureDigits ? toFixed(ensureDigits) : valueNoOp;
+
+    useEffect(() => {
+        onChange(innerValue);
+    }, [innerValue, onChange]);
+
+    const handleChange = (v: string | undefined) => {
+        if (v === undefined) {
+            setInnerValue(0);
+            return;
+        }
+
+        const parser = Number.isInteger(step) ? parseInt : parseFloat;
```

Comment:
There is some weird behaviour where I cannot create a decimal number from scratch.

Reply:
Yes, I tried to emulate the InlineNumber, but that seems to not really work, so I've ported it over instead.

Reply:
I've fixed that now.